errbot/utils.py

```diff
@@ -160,11 +162,11 @@ def find_roots(path, file_sig='*.plug'):
                     log.debug('Ignore %s.', dir_to_add)
                     break
             else:
-                roots.add(dir_to_add)
-    return roots
+                roots.append(dir_to_add)
+    return list(collections.OrderedDict.fromkeys(roots))
```

Comment:
What about letting the set in the implementation and just return the sorted list at the end?

Reply:
I'm not sure what you mean here, could you explain a bit more?

Reply:
I think he's tying to say that this maybe could be simplified by doing the mutation / sorting at the return line, instead of changing the type of `roots`. My guess is that a set ensures we have unique entries in the resolution of `roots`.

Maybe something like this would work:
```
roots = set()
for ...
return sorted(roots)
```

On another note, i never realized this but it seems like a `set()` already sorts entries?
```
In [17]: roots = {'g', 'c', 'b', 'd', 'a'}                                                                                                                                   

In [18]: roots                                                                                                                                                               
Out[18]: {'a', 'b', 'c', 'd', 'g'}

In [19]: roots = {'/g', '/c', '/b', '/d', '/a'}                                                                                                                              

In [20]: roots                                                                                                                                                               
Out[20]: {'/a', '/b', '/c', '/d', '/g'}

```

Reply:
Either way, your changes does make the clashing deterministic.